aspnetcore/performance/caching/memory/sample/RPcache/Startup.cs

```diff
@@ -0,0 +1,47 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using RPcache.Services;
+
+namespace RPcache
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        #region snippet
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+
+            services.AddSingleton<MyMemoryCache>();
```

Comment:
I think we want to explain why we add a separate wrapper class in this scenario. The reason is that we don't know whether other components may use a cache with a size limit and how the units are defined. We create our own class and resolve it in our own services so we have a consistent definition of size unit.

Reply:
`MyMemoryCache` is created as an independent memory cache for components that are aware of this szize limited cache and know how to set cache entry size appropriately.